### PR TITLE
Add consent management reset functionality to properly disable TCF/GPP modules

### PR DESCRIPTION
**Pull request by pm-nitin-shirsat**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [x] Other (Enhancement)

## Description of change
This PR addresses issue #13651 where TCF/GPP consent management modules are not fully disabled when switching consent configurations. The system now properly resets consent handler state and removes CMP event listeners when modules are disabled.
- Clean module state transitions when switching consent configurations
- Prevents memory leaks from active CMP listeners
- Eliminates false warnings from disabled modules
- Backward compatible - no breaking changes

### Key Changes
- ConsentHandler: Added CMP API tracking (setCmpApi, getCmpApi) and listener management (setCmpListenerId, getCmpListenerId)
- Reset Method: Enhanced to remove CMP event listeners and clear references
- TCF Module: Tracks CMP API and listener ID during initialization
- GPP Module: Tracks CMP API and listener ID in event callbacks
- Unit Tests: Comprehensive coverage including edge cases (null, undefined, 0 values)

## Other information
Refer Issue: #13651

### CODE REVIEW

File: src/consentHandler.ts [hunk 1]
```diff
@@ -108,16 +147,41 @@ export class ConsentHandler<T> {
...
+        callback: this.resetCmpApis.bind(this),
+        parameter: this.getCmpListenerId(),
+      };
+
+      // Call the method that subclasses will override
+      this.addApiVersionToParams(params);
+
+      this.getCmpApi()(params);
+    }
+  }
```
Comment (dgirardi) [inline src/consentHandler.ts]: this should be done in the consentManagement modules, where all other CMP related logic is. To share between TCF and GPP you could extract it into `libraries`; but since it's superfluous if you're not using particular modules, it doesn't need to be in core.

Comment (pm-nitin-shirsat) [reply src/consentHandler.ts]: @dgirardi - Moved CMP related logic separate to a file.

File: src/consentHandler.ts [hunk 2]
```diff
@@ -132,16 +139,21 @@ class UspConsentHandler extends ConsentHandler<ConsentDataFor<typeof CONSENT_USP
 }
 
 class GdprConsentHandler extends ConsentHandler<ConsentDataFor<typeof CONSENT_GDPR>> {
-  hashFields = ['gdprApplies', 'consentString']
+  hashFields = ["gdprApplies", "consentString"];
+  /**
+   * Remove CMP event listener using CMP API
+   */
```
Comment (dgirardi) [inline src/consentHandler.ts:145]: stray comment

**Comment by patmmccann** [comment]
@pm-nitin-shirsat please document this new feature
